api/scpca_portal/test/models/test_job.py

```diff
@@ -66,6 +75,163 @@ def test_submit_job_failure(self, mock_batch_submit_job):
+        saved_job = Job.objects.get(pk=submitted_job.pk)
+        self.assertEqual(saved_job, submitted_job)
+
+        # Set up mock for get_jobs for RUNNING
+        mock_batch_get_jobs.return_value = [
+            {
+                "status": "RUNNING",
+                "statusReason": "Job RUNNING",
+            }
+        ]
+
+        success = submitted_job.sync_state()
+        mock_batch_get_jobs.assert_called()
+        self.assertTrue(success)  # Synced but no update in the db
+
+        # The job should remain unchanged and unsaved
+        saved_job = Job.objects.get(pk=submitted_job.pk)
+        self.assertEqual(saved_job, submitted_job)
+
+        # Set up mock for get_jobs with TERMINATED
+        mock_batch_get_jobs.return_value = [
+            {
+                "status": "FAILED",
+                "statusReason": "Job FAILED",
+                "isTerminated": True,
+            }
+        ]
+
+        success = submitted_job.sync_state()
+        mock_batch_get_jobs.assert_called()
+        self.assertTrue(success)  # Synced and updated the db
+
+        # Job should be updated and saved with correct field values
+        saved_job = Job.objects.get(pk=submitted_job.pk)
+        self.assertEqual(saved_job.state, JobStates.TERMINATED.name)
+        self.assertIsInstance(saved_job.terminated_at, datetime)
+        self.assertIsNone(saved_job.failure_reason)
+
+        # Set up mock for get_jobs with FAILED
+        submitted_job.state = JobStates.SUBMITTED.name
+        mock_batch_get_jobs.return_value = [
+            {
+                "status": "FAILED",
+                "statusReason": "Job FAILED",
+            }
+        ]
+
+        success = submitted_job.sync_state()
+        mock_batch_get_jobs.assert_called()
+        self.assertTrue(success)  # Synced and updated the db
+
+        # Job should be updated and saved with correct field values
+        saved_job = Job.objects.get(pk=submitted_job.pk)
+        self.assertEqual(saved_job.state, JobStates.COMPLETED.name)
+        self.assertEqual(saved_job.failure_reason, "Job FAILED")
+        self.assertIsInstance(saved_job.completed_at, datetime)
+
+    @patch("scpca_portal.batch.get_jobs")
+    def test_bulk_sync_state(self, mock_batch_get_jobs):
+        # Set up mock for get_jobs with all AWS Batch job statuses including terminated FAILED job
+        mock_batch_get_jobs.return_value = JobFactory.get_mock_aws_batch_jobs(
+            [
+                {"jobId": f"{self.mock_batch_job_id}-{i}", "status": status}
+                for i, status in enumerate(
+                    self.aws_batch_job_status,
+                )
+            ],
+            terminated_job={"jobId": f"{self.mock_batch_job_id}-7", "status": "FAILED"},
+        )
+        JobFactory.get_mock_jobs(
+            [
+                {"batch_job_id": f"{self.mock_batch_job_id}-{i}", "state": JobStates.SUBMITTED.name}
+                for i in range(0, 8)
+            ]
+        )
```

Comment:
```suggestion
        jobs = [JobFactory() for _ in range(7)]
                
        mock_response = [{"jobId": job.aws_batch_id} for job in jobs]
        
        mock_response[0]["status"] = "PENDING"
        # .... all other jobs
        mock_response[4]["status"] = "TERMINATED"
        
        mock_batch_get_jobs.return_value = mock_response
        
```

You should be able to use `factory.Sequence(lambda n: f"batch-id{n}")` to dynamically define the `aws_batch_job_id` attributes